js/messaging.js

```diff
@@ -233,12 +238,42 @@ function openURL({
       url.replace(/%2F.*/, '*').replace(/#.*/, '') :
       url.replace(/#.*/, '');
 
-  return queryTabs({url: urlQuery, currentWindow}).then(maybeSwitch);
+  return manageMatch || editMatch ? queryTabs().then(maybeSwitch) :
+  queryTabs({url: urlQuery, currentWindow}).then(maybeSwitch);
 
   function maybeSwitch(tabs = []) {
     const urlWithSlash = url + '/';
     const urlFF = FIREFOX && url.replace(/%2F/g, '/');
-    const tab = tabs.find(({url: u}) => u === url || u === urlFF || u === urlWithSlash);
+    const urlOptions = manageMatch ? URLS.ownOrigin + 'manage.html#stylus-options' : null;
+    const urlManage = manageMatch ? URLS.ownOrigin + 'manage.html' : null;
+    const tab = tabs.find(({url: u}) => u === url || u === urlFF || u === urlWithSlash ||
+                          u === urlOptions || u === urlManage);
+    if (!tab && prefs.get('openEditInWindow') && chrome.windows && editMatch) {
+      chrome.windows.create(
+        Object.assign({
+          url: url
+        }, prefs.get('windowPosition', {}))
+      );
+      return;
+    }
+    if (manageMatch) {
+      if (tab) {
+        const toggleOptions = url === urlOptions ? 'options-open' : 'options-close';
+        chrome.tabs.sendMessage(tab.id, {
```

Comment:
>it will work with all pages

What "all pages" does it need to work with though? If a manager tab is open and you click one of the options links, it focuses the open manage tab and sends a message to open options, and vice-versa.

>Can we use chrome.tabs.update to update the URL?

I suppose. I don't really see how it matters.